docs/CommunicationIOS.md

```diff
@@ -0,0 +1,221 @@
+
+RCTRootView *rootView = [[RCTRootView alloc] initWithBridge:bridge
+                                                 moduleName:@"ImageBrowserApp"
+                                          initialProperties:props];
+```
+
+```
+'use strict';
+
+var React = require('react-native');
+  var {
+  View,
+  Image
+} = React;
+
+class ImageBrowserApp extends React.Component {
+  renderImage: function(imgURI) {
+    return (
+      <Image source={imgURI} />
+    );
+  },
+  render() {
+    return (
+      <View>
+        {this.props.imageList.map(this.renderImage)}
+      </View>
+    )
+  }
+}
+
+React.AppRegistry.registerComponent('ImageBrowserApp', () => ImageBrowserApp);
+```
+
+`RCTRootView` also provides a read-write property `appProperties`. After `appProperties` is set, the React Native app is re-rendered with new properties. The update is only performed when the new updated properties differ from the previous ones.
+
+```
+NSArray *imageList = @[@"http://foo.com/bar3.png",
+                       @"http://foo.com/bar4.png"];
+
+rootView.appProperties = @{@"images" : imageList};
+```
+
+It is fine to update properties anytime. However, updates have to be performed on the main thread. You use the getter on any thread.
+
+There is no way to update only a few properties at a time. We suggest that you build it into your own wrapper instead.
+
+> ***Note:***
+> Currently, JS functions `componentWillReceiveProps` and `componentWillUpdateProps` of the top level RN component will not be called after a prop update. However, you can access the new props in `componentWillMount` function.
+
+### Passing properties from React Native to native
+The problem exposing properties of native components is covered in detail in [this article](https://facebook.github.io/react-native/docs/native-components-ios.html#properties). In short, export properties with `RCT_CUSTOM_VIEW_PROPERTY` macro in your custom native component, then just use them in React Native as if the component was an ordinary React Native component.
+
+### Limits of properties
+
+The main drawback of cross-language properties is that they do not support callbacks, which would allow us to handle bottom-up data bindings. Imagine you have a small RN view that you want to be removed from the native parent view as a result of a JS action. There is no way to do that with props, as the information would need to go bottom-up.
+
+Although we have a flavor of cross-language callbacks ([described here](https://facebook.github.io/react-native/docs/native-modules-ios.html#callbacks)), these callbacks are not always the thing we need. The main problem is that they are not intended to be passed as properties. Rather, this mechanism allows us to trigger a native action from JS, and handle the result of that action in JS.
+
+Ideally, some day we'll fully support converting JS functions to native (and vice versa), so we can pass functions transparently between native and React Native.
```

Comment:
I wouldn't put this in the documentation like this, as we're not sure we'll ever want to support this. I think most of what you describe here can be done with native modules (+ callbacks), even though it's not super flexible.